### PR TITLE
Implement stylelint and precommit hook

### PR DESCRIPTION
**Pull request by zcuric**
### This PR
- [x] Implements `stylelint`
- [x] Adds `precommit` hook for linting staged files
- [x] Fixes lint issues in the code base

### CODE REVIEW

File: src/teaching-element/index.vue [hunk 1]
```diff
@@ -77,17 +77,16 @@ export default {
...
-     -khtml-user-select: none; /* Konqueror HTML */
-       -moz-user-select: none; /* Firefox */
-        -ms-user-select: none; /* Internet Explorer/Edge */
-            user-select: none; /* Non-prefixed version, currently
-                                  supported by Chrome and Opera */
+  -webkit-user-select: none; /* Safari */
+  -khtml-user-select: none; /* Konqueror HTML */
+  -moz-user-select: none; /* Firefox */
+  -ms-user-select: none; /* Internet Explorer/Edge */
+  user-select: none; /* Non-prefixed version, currently supported by Chrome and Opera */
```
Comment (vladimyr) [inline src/teaching-element/index.vue]: This is exactly why we have autoprefixer. 🙃

**Comment by vladimyr** [comment]
[`Lint everuthing :trollface`](https://github.com/ExtensionEngine/tailor-teaching-elements/commit/40db2fcd7b9960643539674695a82a94ce4a5ebc) 👌😂

File: src/teaching-element/index.vue (hunk 1 above)
Comment (zcuric) [reply src/teaching-element/index.vue]: This was linted, boi. It was already there. 💋

**Comment by zcuric** [comment]
> [`Lint everuthing :trollface`](https://github.com/ExtensionEngine/tailor-teaching-elements/commit/40db2fcd7b9960643539674695a82a94ce4a5ebc) 👌😂

I'm nepismen.

**Comment by vladimyr** [comment]
I just assume your commit msg editor to be vim 😂

**Comment by zcuric** [comment]
@vladimyr Per our discussion, show we then close this and YOLO, or merge it and play safe?

**Comment by zcuric** [comment]
@kronicker this can be merged.

**Pull request closed without merging**